catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/DavAlterationObserver.java

```diff
@@ -0,0 +1,306 @@
+            final DavEntry[] children = doListFiles(rootEntry.getLocation(), rootEntry);
+            rootEntry.setChildren(children);
+        } catch (IOException e) {
+            // probably means root location was inaccessible
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Final processing.
+     *
+     * @throws Exception if an error occurs
+     */
+    public void destroy() throws Exception {
+    }
+
+    /**
+     * Check whether the file and its chlidren have been created, modified or deleted.
+     */
+    void checkAndNotify(Sardine sardine) {
+        this.sardine = sardine;
+        if (rootEntry.remoteExists(sardine)) {
+            checkAndNotify(rootEntry, rootEntry.getChildren(), listFiles(rootEntry.getLocation()));
+        } else if (rootEntry.isExists()) {
+            // would remove all metacards if connectivity is lost, but tough to distinguish
+            // from the hierarchy not existing remotely
+            checkAndNotify(rootEntry, rootEntry.getChildren(), EMPTY_RESOURCES);
+        } else {
+            // Didn't exist and still doesn't
+        }
+    }
+
+    /**
+     * Compare two file lists for files which have been created, modified or deleted.
+     *
+     * @param parent   The parent entry
+     * @param previous The original list of files
+     * @param incoming The current list of files
+     */
+    private void checkAndNotify(final DavEntry parent, final DavEntry[] previous,
+            final DavResource[] incoming) {
+        int c = 0;
+        final DavEntry[] current =
+                incoming.length > 0 ? new DavEntry[incoming.length] : DavEntry.getEmptyEntries();
+        // this loop relies on lexicographical sorting, so compareTo < 0 will never be hit
+        for (final DavEntry entry : previous) {
+            // new entries, sorted into place between existing entries
+            while (c < incoming.length && entry.getLocation()
+                    .compareTo(DavEntry.getLocation(incoming[c].getName(), entry.getParent()))
+                    > 0) {
+                current[c] = createFileEntry(parent, incoming[c]);
+                doCreate(current[c]);
+                c++;
+            }
+            // check if existing entry has been modified
+            if (c < incoming.length && entry.getLocation()
+                    .compareTo(DavEntry.getLocation(incoming[c].getName(), entry.getParent()))
+                    == 0) {
+                doMatch(entry, incoming[c]);
+                checkAndNotify(entry, entry.getChildren(), listFiles(entry.getLocation()));
+                current[c] = entry;
+                c++;
+
+            } else {
+                // entry no longer exists, so delete it
+                checkAndNotify(entry, entry.getChildren(), EMPTY_RESOURCES);
+                doDelete(entry);
+            }
+        }
+        // create anything remaining at the end
+        for (; c < incoming.length; c++) {
```

Comment:
I think I like the for loop better, keeps the step and condition in the same place.